en/manual/scripts/serialization.md

```diff
@@ -0,0 +1,145 @@
+﻿# Serialization
+
+<span class="label label-doc-level">Beginner</span>
+<span class="label label-doc-audience">Programmer</span>
+
+The editor and serialization system uses four attributes to determine what is serialized and visible in the editor.
+
+### [DataContractAttribute](xref:Stride.Core.DataContractAttribute)
+Adding this attribute to your `class` or `struct` notifies the serializer and the editor that it should
+show fields and properties of that type, and serialize the data it contains with the scenes or assets that might include it.
+```cs
+[DataContract(Inherited = true)]
+public class MySerializedClass
+{
+    public float MyValue;
+}
+
+// 'DataContract' is inherited above. You don't need to specify it for a derived class.
+public class MyDerivedSerializedClass : MySerializedClass
+{
+    public string MyName;
+}
+```
+
+> [!Note]
+> Your IDE may wrongfully add `System.Runtime.Serialization` to your list of `using` when adding `DataContract`.
+> They are not interchangeable.
+> Make sure that your `DataContract` comes from `Stride.Core`, specifying the namespace explicitly like shown above if necessary.
+
+### [DataMemberAttribute](xref:Stride.Core.DataMemberAttribute)
+This notifies the editor and serializer that the property or field on this [DataContract](#datacontractattribute)'ed
+`class` or `struct` should be serialized.
+Note that you can omit this attribute for most public fields and properties, they will be included by default,
+see [Fields](#fields) and [Properties](#properties) for specifics.
+```cs
+[Stride.Core.DataContract]
+public class MySerializedClass
+{
+    [Stride.Core.DataMember]
+    internal float MyValue;
+}
+```
+
+### [DataAliasAttribute](xref:Stride.Core.DataAliasAttribute)
+Can be used to ensure you do not break previously serialized data whenever you have to change how that member is named in your source.
```

Comment:
I think we need to mention that this only applies for the editor. There is no binary compatibility with  the previous class for compiled assets.

Reply:
What do you mean exactly, that the alias is not checked at runtime if they somehow loaded a previous version of the asset hosting that field ?

Reply:
What I mean is that the aliasing is a purely YAML based feature.
In binary serialization the serializer makes an arbitrary decision about field ordering when serializing into a stream of bytes. There is some fingerprinting mechanism which detects the type changed and prevents deserialization from another version of the type.

Reply:
Added a note, let me know what you think.

Reply:
Looks good